x-pack/test/apm_api_integration/tests/storage_explorer/storage_explorer_summary_stats.spec.ts

```diff
@@ -0,0 +1,149 @@
+  const apmApiClient = getService('apmApiClient');
+  const synthtraceEsClient = getService('synthtraceEsClient');
+
+  const start = new Date('2021-01-01T00:00:00.000Z').getTime();
+  const end = new Date('2021-01-01T00:15:00.000Z').getTime() - 1;
+  const goServiceName = 'opbeans-go';
+  const nodeServiceName = 'opbeans-node';
+
+  async function callApi(
+    overrides?: RecursivePartial<
+      APIClientRequestParamsOf<'GET /internal/apm/storage_explorer_summary_stats'>['params']
+    >
+  ) {
+    return await apmApiClient.monitorIndicesUser({
+      endpoint: 'GET /internal/apm/storage_explorer_summary_stats',
+      params: {
+        query: {
+          indexLifecyclePhase: IndexLifecyclePhaseSelectOption.All,
+          probability: 1,
+          environment: 'ENVIRONMENT_ALL',
+          start: new Date(start).toISOString(),
+          end: new Date(end).toISOString(),
+          kuery: '',
+          ...overrides?.query,
+        },
+      },
+    });
+  }
+
+  registry.when(
+    'Storage explorer summary stats when data is not loaded',
+    { config: 'basic', archives: [] },
+    () => {
+      it('handles empty state', async () => {
+        const { status, body } = await callApi();
+
+        expect(status).to.be(200);
+        expect(body.tracesPerMinute).to.be(0);
+        expect(body.numberOfServices).to.be(0);
+        expect(body.estimatedSize).to.be(0);
+        expect(body.dailyDataGeneration).to.be(0);
+      });
+    }
+  );
+
+  registry.when('Storage explorer summary stats', { config: 'basic', archives: [] }, () => {
+    describe('when data is loaded', () => {
+      before(async () => {
+        const serviceGo = apm
+          .service({ name: goServiceName, environment: 'production', agentName: 'go' })
+          .instance('instance-go');
+
+        const serviceNode = apm
+          .service({ name: nodeServiceName, environment: 'dev', agentName: 'node' })
+          .instance('instance-node');
+
+        await synthtraceEsClient.index([
+          timerange(start, end)
+            .interval('1m')
+            .rate(1)
+            .generator((timestamp) =>
+              serviceGo
+                .transaction({ transactionName: 'GET /api/product/list' })
+                .duration(2000)
+                .timestamp(timestamp)
+            ),
+          timerange(start, end)
+            .interval('1m')
+            .rate(1)
+            .generator((timestamp) =>
+              serviceNode
+                .transaction({ transactionName: 'GET /api/users/list' })
+                .duration(2000)
+                .timestamp(timestamp)
+            ),
+        ]);
+      });
+
+      after(() => synthtraceEsClient.clean());
+
+      it('returns correct summary stats', async () => {
+        const { status, body } = await callApi();
+
+        expect(status).to.be(200);
+        expect(body.numberOfServices).to.be(2);
+        expect(Math.round(body.tracesPerMinute)).to.be(2);
```

Comment:
you could use https://github.com/cauemarcondes/kibana/blob/apm-labs-telemetry/x-pack/test/apm_api_integration/utils.ts#L12:L12. but it's up to you.

Reply:
Updated in https://github.com/elastic/kibana/pull/141391/commits/b8679d235120a186f521b07891eba768e504adf6